### PR TITLE
Feat: Blog deletion endpoint added

### PR DESCRIPTION
**Pull request by codeReaper0**
### Title
**Fix/Delete Blog Endpoint: Handle Invalid UUID and Other Edge Cases**
​

## Description
**Updated the delete_blog endpoint in the FastAPI application to handle various edge cases and improve error handling:**

1. UUID Validation: Added validation to check for proper UUID format before attempting to delete a blog.
2. Deletion Check: Added logic to check if the blog is already deleted and return an appropriate error if it is.
3. Error Handling: Improved error handling for cases where the blog does not exist or where a server error occurs during the commit operation.
4. Test Coverage: Added comprehensive test cases to cover all edge cases, including valid and invalid UUIDs, non-existent blogs, already deleted blogs, and server errors.

​

## Related Issue (Link to issue ticket)
[**Issue #67 [FEAT] Blog Post Deletion Endpoint ](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/67)**

​

## Motivation and Context

The changes are required to ensure that the delete_blog endpoint:

- Correctly validates UUIDs to prevent errors related to invalid input.
- Handles the case where a blog has already been marked as deleted.
- Provides appropriate error responses for cases where the blog is not found or if there is a server-side issue during deletion.
- Ensures robustness and reliability of the endpoint through comprehensive testing.
- 

​

## How Has This Been Tested?

Unit Tests: Added tests for the delete_blog endpoint to cover:
- Successful deletion of a blog with a valid UUID.
- Attempt to delete a blog that has already been marked as deleted.
- Attempt to delete a non-existent blog.
- Attempt to delete with an invalid UUID.
- Handling server errors during the commit operation.

**Testing Environment:**
Database: Configured a separate test database to ensure isolation of test cases.
Testing Framework: Used pytest for running the tests.

**Tests Run:**
- test_delete_blog_valid_uuid: Valid UUID for successful deletion.
- test_delete_blog_already_deleted: Blog that has already been deleted.
- test_delete_blog_not_exist: Non-existent blog.
- test_delete_blog_invalid_uuid: Invalid UUID format.
- test_delete_blog_server_error: Simulated server error during commit.

​

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from api.v1.schemas.blog import DeleteBlogResponseSchema
+from api.utils.dependencies import get_current_admin
+import logging
+import uuid
+
+
+blog = APIRouter(prefix="/api/v1/blogs", tags=["Blogs"])
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
```
Comment (Dev-wonderful) [inline api/v1/routes/blog.py:14]: You have made a mistake with the `prefix`, it should be `/blogs` only

Also you are like the 3rd person to implement logging you guys should implement it properly. You can turn it to a utility for everyone to use

File: api/v1/routes/blog.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    try:
+        uuid.UUID(id)
+        return True
+    except ValueError:
+        return False
+
+
+@blog.delete("/{id}", response_model=DeleteBlogResponseSchema)
+async def delete_blog(id: str, db: Session = Depends(get_db), user: str = Depends(get_current_admin)):
+    if not validate_uuid(id):
```
Comment (Dev-wonderful) [inline api/v1/routes/blog.py:26]: Isn't `get_current_admin` returning a `User` object, why are you using `str` as the type

File: tests/v1/test_delete_blog.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Base.metadata.create_all(bind=engine)
+
+
+def override_get_db():
+    try:
+        db = SessionLocal()
+        yield db
+    finally:
+        db.close()
+
```
Comment (Dev-wonderful) [inline tests/v1/test_delete_blog.py:38]: Mock the database

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/routes/blog.py (hunk 1 above)
Comment (codeReaper0) [reply api/v1/routes/blog.py]: I have removed the logger

File: api/v1/routes/blog.py (hunk 2 above)
Comment (codeReaper0) [reply api/v1/routes/blog.py]: I have adjusted it

File: tests/v1/test_delete_blog.py (hunk 3 above)
Comment (codeReaper0) [reply tests/v1/test_delete_blog.py]: I have mocked the db
